src/test/java/com/list/todo/it/TagTest.java

```diff
@@ -0,0 +1,450 @@
+            UserPrincipal userPrincipal = new UserPrincipal();
+            userPrincipal.setId(1L);
+            userPrincipal.setUsername("username");
+            userPrincipal.setPassword(new BCryptPasswordEncoder().encode("password"));
+
+            return userPrincipal;
+        }
+    };
+
+    @Before
+    public void beforeMethod() {
+        mockMvc = MockMvcBuilders
+                .standaloneSetup(tagControllerMock)
+                .setCustomArgumentResolvers(
+                        putAuthenticationPrincipal,
+                        new HandlerMethodArgumentResolver() {
+
+                            @Override
+                            public boolean supportsParameter(
+                                    MethodParameter parameter) {
+                                return parameter.getParameterType().equals(
+                                        Pageable.class);
+                            }
+
+                            @Override
+                            public Object resolveArgument(
+                                    MethodParameter parameter,
+                                    ModelAndViewContainer mavContainer,
+                                    NativeWebRequest webRequest,
+                                    WebDataBinderFactory binderFactory) {
+
+                                return new PageRequest(0, 50);
+                            }
+                        })
+                .build();
+    }
+
+    @Test
+    public void getMyTags_ReturnsAListOfTags() throws Exception {
+        //arrange
+        when(tagServiceMock.getTagsByOwnerId(1L)).thenReturn(getListOfTags());
+
+        //act
+        MvcResult result = this.mockMvc.perform(get("/api/tags"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andReturn();
+        //assert
+        assertEquals(getTagsFromJsonResponse(result.getResponse().getContentAsString(),
+                getListOfTags().size()).hashCode(), getListOfTags().hashCode());
+
+    }
+
+    @Test
+    public void getMyTagsWithTodoListId_OnExistentTagsIds_ReturnsAListOfTagTaskKeys() throws Exception {
+        //arrange
+
+        when(tagTaskKeyServiceMock.getMyTaggedTask(any(UserPrincipal.class), any(Pageable.class), any()))
+                .thenReturn(getListOfTaggedTask());
+
+        //act
+        MvcResult result = this.mockMvc.perform(get("/api/tags/myTagTaskKeys?tagId=6"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andReturn();
+
+        //assert
+        assertEquals(new HashSet<>(getTagTaskKeyFromJsonResponse(result.getResponse().getContentAsString(),
+                getListOfTaggedTask().size())).hashCode(), getListOfTaggedTask().hashCode());
+    }
+
+    @Test
+    public void addTag_ReturnsAObjectOfTag() throws Exception {
+        //arrange
+        when(tagServiceMock.addTag(any(), any())).thenReturn(Optional.of(new Tag("Home", 1L, "ff")));
```

Comment:
You must check whether data you pass to tagService is correct.
To make it simpler you can serialize tagInput to JSON string and pass result as .content() argument, instead of hardcoding json string "manually"